AutomationSelenium/Pages/Components/ProfilePageOverview/AddEditLanguageComponent.cs

```diff
@@ -0,0 +1,174 @@
+
+        }
+
+        public void renderLanguageInputComponents()
+        {
+            try
+            {
+                languageTab = driver.FindElement(By.XPath("//a[normalize-space()='Languages']"));
+                languageTxtBox = driver.FindElement(By.Name("name"));
+                levelTxtBox = driver.FindElement(By.Name("level"));
+                addBtn = driver.FindElement(By.XPath("//input[@value='Add']"));
+                cancelBtn = driver.FindElement(By.XPath("//input[@value='Cancel']"));
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex);
+            }
+        }
+        public void renderLanguageTestComponent()
+        {
+            addedNewLanguage = driver.FindElement(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/tbody[last()]/tr/td[1]"));
+        }
+        public void renderUpdateInputComponents()
+        {
+            languageEditTxtBox = driver.FindElement(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/tbody[1]/tr/td/div/div[1]/input"));
+            levelEditTxtBox = driver.FindElement(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/tbody[1]/tr/td/div/div[2]/select"));
+
+        }
+        public void renderUpdateComponent()
+        {
+            updateBtn = driver.FindElement(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/tbody[1]/tr/td/div/span/input[1]"));
+        }
+        public void renderUpdateTestComponent()
+        {
+            updatedLanguage = driver.FindElement(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/tbody[1]/tr/td[1]"));
+        }
+        public void renderMessageComponent()
+        {
+            messageBox = driver.FindElement(By.XPath("//div[@class='ns-box-inner']"));
+        }
+        public void clearExistingdata()
+        {
+            try
+            {
+                IWebElement deleteButton = driver.FindElement(By.XPath("//div[2]/div/div[2]/div/table/tbody/tr/td[3]/span[2]/i"));
+                var deleteButtons = driver.FindElements(By.XPath("//div[2]/div/div[2]/div/table/tbody/tr/td[3]/span[2]/i"));
+                foreach (var button in deleteButtons)
+                {
+                    button.Click();
+                }
+
+            }
+            catch (NoSuchElementException)
+            {
+                Console.WriteLine("no items to delete");
+            }
+
+        }
+        public void addNewLanguage(LanguageTestModel data)
+        {
+            Wait.WaitToBeClickable(driver, "XPath", "//div[@class='ui bottom attached tab segment active tooltip-target']//div[contains(@class,'ui teal button')][normalize-space()='Add New']", 12);
```

Comment:
have you thought about state, ensure that you have reset state, (there should be not skills present, if skills are present you should delete all and the add) this will ensure that there is no false positive when you execute the test.